doc/meson.build

```diff
@@ -0,0 +1,58 @@
+valadoc = find_program('valadoc', required: false)
+gd_scan = find_program('gtkdoc-scan', required: false)
+
+if not valadoc.found()
+    message('valadoc is not installed. building valadocs is not possible.')
```

Comment:
I prefer having an option in `meson_options.txt` specifying if this should be enabled than having an hidden and soft fallback, this helps packagers to do things right ;)

Reply:
I pushed a commit containing this change, but it breaks running "ninja install" when documentation is enabled. meson/ninja runs the custom targets for the docs twice (on "ninja" and on "ninja install", and I couldn't figure out how to make them run only once.